### ISSUE TITLE
Fix the failing tests after tt-mlir uplift

### ISSUE THREAD

**Issue by uazizTT** [tenstorrent/tt-xla#41]
Three tests are failing due to the same error after uplifting tt-mlir, fix the issue and then enable these tests. 

```
FAILED tests/TTIR/test_basic_ops.py::test_dot_general_op - RuntimeError: Failed to run VHLO Legalization pass pipeline.
FAILED tests/TTIR/test_mnist.py::test_matmul - RuntimeError: Failed to run VHLO Legalization pass pipeline.
FAILED tests/TTIR/test_mnist.py::test_matmul_with_bias - RuntimeError: Failed to run VHLO Legalization pass pipeline.
```

**Comment by LPanosTT** [comment]
Looking at the latest version of stablehlo, it seems like there does not exist a conversion pattern for the op `vhlo.dot_general_v1` to `stablehlo.dot_general` (see [here](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/VhloLegalizeToStablehlo.cpp)). However, there is a conversion pattern for `vhlo.dot_general_v2`. Unfortunately jax generates `vhlo.dot_general_v1` ops for these tests.

Furthermore, stablehlo does have VhloToVersionPass which looks like it can upgrade V1 ops to V2 ops. However, very few ops have pattern rewires, and dot_general_v1 is not one of them. They do seem to have a pattern declaration for that though in a [tablegen](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/VhloToVersionPatterns.td) file. I suppose they just have not implemented it:

**Comment by mrakitaTT** [comment]
Thank you for all the info @LPanosTT!! I'll investigate further.

**Comment by LPanosTT** [comment]
@mrakitaTT I have been able to produce a stablehlo module for the mnist matmul test following the instructions here: https://openxla.org/stablehlo/tutorials/jax-export#export_jax_model_to_stablehlo_using_jaxexport

Maybe we just need to replicate what passes that is using somehow.

**Comment by mrakitaTT** [comment]
I've managed to fix this by adding `-vhlo-to-version` pass before `-vhlo-legalize-to-stablehlo`, in order to update VHLO to latest version. That converts v1 ops to v2 ops, and then `-vhlo-legalize-to-stablehlo` successfully converts `vhlo.dot_general_v2`.

Thank you @LPanosTT for the idea! When I looked into tablegen file that you've mentioned I've realized that they autogenerate conversions from that tablegen file, that is why you couldn't find the implementation 🙂 

Will send fix on PR tomorrow. FYI @vladimirjovanovicTT
